src/Rules/FileOrStringRule.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace EscolaLms\Files\Rules;
+
+use Illuminate\Contracts\Validation\Rule;
+use Illuminate\Http\UploadedFile;
+use Illuminate\Support\Facades\Storage;
+use Illuminate\Support\Facades\Validator;
+
+class FileOrStringRule implements Rule
+{
+    private array $fileRules;
+    private ?string $pathPrefix;
+    private string $message;
+
+    public function __construct(array $fileRules, string $pathPrefix = null)
+    {
+        $this->fileRules = $fileRules;
+        $this->pathPrefix = $pathPrefix;
+    }
+
+    public function passes($attribute, $value): bool
+    {
+        if (is_a($value, UploadedFile::class) && !$this->isValidFile($attribute, $value)) {
+            return false;
+        }
+
+        if (is_string($value)) {
+            if (!Storage::exists($value)) {
+                $this->message = __('File not exist');
+                return false;
+            }
+
+            if ($this->pathPrefix && !str_starts_with($value, $this->pathPrefix)) {
```

Comment:
FYI str_starts_with is php8+, so in php7.4 this works only because symfony/polyfill-php80 is installed